src/server/api/routes/users/v2/wallet.js

```diff
@@ -0,0 +1,33 @@
+const { Router } = require('express');
+const { walletUserCollection } = require('../../../../modules/firebase');
+const { GET_WALLET_API_SECRET } = require('../../../../config/config');
+
+const router = Router();
+
+router.get('/wallet', async (req, res, next) => {
+  try {
+    const { email, key } = req.query;
+    if (!email) {
+      res.status(400).send('MISSING_EMAIL');
+      return;
+    }
+    if (key !== GET_WALLET_API_SECRET) {
```

Comment:
We might want to make an extra guard that throws 403 on `GET_WALLET_API_SECRET` is not set, in case of deployment error that we did not set a `GET_WALLET_API_SECRET`